### ISSUE TITLE
Replace iron-list with vaadin-grid

### ISSUE THREAD

**Issue by tomivirkki** [vaadin/vaadin-combo-box#32]
- Replace the element in vaadin-combo-box
- Make all the existing APIs work with vaadin-grid
- Make sure resizing / keyboard navigation works ok
 -  Try to make the vaadin-grid built-in focus work first, if it seems not doable, use a similar approach as with iron-list currently (change selection on key press and perhaps use a rowclassgeenrator for highlighting the vaadin-combo-box selection)

**Comment by Saulis** [comment]
Some conclusions on #33: 
- Keyboard navigation works using `rowClassGenerator` + `grid.scrollToRow(focusedIndex)` approach nicely
- Correct sizing of `<vaadin-grid>` needs some tweaking. (will be seen how it's going to work after #21)

**Comment by Saulis** [comment]
Work done in `feature/vaadin-grid` branch:
- Edge has overflow issues on demos
- Edge keyboard navigation tests are failing
- Grid native keyboard navigation needs to be disabled (or native focus styles removed)

**Comment by jounik** [comment]
The upcoming Material Design styles (#35) will need to be refactored to use vaadin-grid also.

**Comment by jounik** [comment]
If the sizing is clearly a vaadin-grid issue, it should be handled separately.

**Comment by ktiedt** [comment]
Howdy, just chiming in because this sounds like a major overkill for what should be a simple list view... from a user perspective this sounds extremely heavy weight if combbox depends on grid.

**Comment by Saulis** [comment]
@ktiedt yes you're right, we've also started to realise the same thing during the past months. The original motivation for us was to reuse the lazy loading functionality of `<vaadin-grid>` in `<vaadin-combo-box>` but at this point a more viable option is most likely to implement that on top of `<iron-list>`. 

Closing this issue as it looks like we're dropping the approach of using `<vaadin-grid>`.

**Comment by jounik** [comment]
Well, the idea was to avoid doing duplicate work on two different elements, for features such as lazy loading, columns, filtering, sorting etc. This would have also forced us to try and optimize the grid as much as possible. But for the current version and it’s features, using vaadin-grid does sound like an overkill, I agree.
